elide-integration-tests/src/test/groovy/com/yahoo/elide/tests/SortIT.groovy

```diff
@@ -280,49 +280,70 @@ class SortIT extends AbstractIntegrationTestInitializer {
+        def firstChapter = chapters.get(0)
+        final String firstChapterBookId = firstChapter["relationships"]["books"]["data"].get(0)["id"].asText()
         Assert.assertEquals(firstChapterBookId, "2")
-        final String firstChapterTitle = firstChapter.get("attributes").get("title").asText()
+        final String firstChapterTitle = firstChapter["attributes"]["title"].asText()
         Assert.assertEquals(firstChapterTitle, "Dobby's Warning")
     }
 
     @Test
     public void testSortOnMultipleJoinEntityAttributes() {
         def result = mapper.readTree(RestAssured.get("/chapter?sort=-books.genre,books.authors.name,-title&page[limit]=3").asString())
-        final JsonNode chapters = result.get("data")
+        def chapters = result["data"]
         Assert.assertEquals(chapters.size(), 3)
-        final JsonNode firstChapter = chapters.get(0)
-        final String firstChapterBookId = firstChapter.get("relationships").get("books").get("data").get(0).get("id").asText()
+        def firstChapter = chapters.get(0)
+        final String firstChapterBookId = firstChapter["relationships"]["books"]["data"].get(0)["id"].asText()
         Assert.assertEquals(firstChapterBookId, "6")
-        final String firstChapterTitle = firstChapter.get("attributes").get("title").asText()
+        final String firstChapterTitle = firstChapter["attributes"]["title"].asText()
         Assert.assertEquals(firstChapterTitle, "What Lucy Found There")
     }
 
     @Test
     public void testSortOnMultipleJoinEntityAttributesWithFilter() {
         def result = mapper.readTree(RestAssured.get(
-                "/chapter?filter[chapter.books.title][prefix]=The&sort=books.genre,books.title,-title&page[limit]=5").asString())
-        final JsonNode chapters = result.get("data")
+                "/chapter?filter[chapter.books.title][prefix]=The&sort=books.genre,books.title,-title&page[limit]=5")
+                .asString())
+        def chapters = result["data"]
         Assert.assertEquals(chapters.size(), 5)
-        final JsonNode firstChapter = chapters.get(0)
-        final String firstChapterBookId = firstChapter.get("relationships").get("books").get("data").get(0).get("id").asText()
+        def firstChapter = chapters.get(0)
+        final String firstChapterBookId = firstChapter["relationships"]["books"]["data"].get(0)["id"].asText()
         Assert.assertEquals(firstChapterBookId, "4")
-        final String firstChapterTitle = firstChapter.get("attributes").get("title").asText()
+        final String firstChapterTitle = firstChapter["attributes"]["title"].asText()
         Assert.assertEquals(firstChapterTitle, "Roast Mutton")
     }
 
+    @Test
+    public void testSortOnMultipleJoinEntityAttributesWithPagination() {
+        def result = mapper.readTree(RestAssured.get(
+                "/chapter?include=books,books.authors&sort=books.authors.name,books.title,title" +
```

Comment:
One idea to address this problem:  Sorting and joins will work just fine assuming the relationship we are traversing is OneToOne or ManyToOne.  Elide could return an error in the other cases.